code/datums/riding.dm

```diff
@@ -0,0 +1,262 @@
+
+/datum/riding
+	var/generic_pixel_x = 0 //All dirs show this pixel_x for the driver
+	var/generic_pixel_y = 0 //All dirs show this pixel_y for the driver, use these vars if the pixel shift is stable across all dir, override handle_vehicle_offsets otherwise.
+	var/next_vehicle_move = 0 //used for move delays
+	var/vehicle_move_delay = 2 //tick delay between movements, lower = faster, higher = slower
+	var/keytype = null
+	var/atom/movable/ridden = null
+
+/datum/riding/proc/handle_vehicle_layer()
+	if(ridden.dir != NORTH)
+		ridden.layer = ABOVE_MOB_LAYER
+	else
+		ridden.layer = OBJ_LAYER
+
+
+//Override this to set your vehicle's various pixel offsets
+//if they differ between directions, otherwise use the
+//generic variables
+/datum/riding/proc/handle_vehicle_offsets()
+	if(ridden.has_buckled_mobs())
+		for(var/m in ridden.buckled_mobs)
+			var/mob/living/buckled_mob = m
+			buckled_mob.setDir(ridden.dir)
+			buckled_mob.pixel_x = generic_pixel_x
+			buckled_mob.pixel_y = generic_pixel_y
+
+//KEYS
+/datum/riding/proc/keycheck(mob/user)
+	if(keytype)
+		if(user.is_holding_item_of_type(keytype))
+			return TRUE
+	else
+		return TRUE
+	return FALSE
+
+
+//BUCKLE HOOKS
+/datum/riding/proc/restore_position(mob/living/buckled_mob,force = 0)
+	if(istype(buckled_mob))
+		buckled_mob.pixel_x = 0
+		buckled_mob.pixel_y = 0
+		if(buckled_mob.client)
+			buckled_mob.client.view = world.view
+
+
+
+
+
+
+//MOVEMENT
+/datum/riding/proc/handle_ride(mob/user, direction)
+	if(user.incapacitated())
+		ridden.unbuckle_mob(user)
+		return
+
+	if(world.time < next_vehicle_move)
+		return
+	next_vehicle_move = world.time + vehicle_move_delay
+	if(keycheck(user))
+		if(!Process_Spacemove(direction) || !isturf(ridden.loc))
+			return
+		step(ridden, direction)
+
+		handle_vehicle_layer()
+		handle_vehicle_offsets()
+	else
+		user << "<span class='notice'>You'll need the keys in one of your hands to drive \the [ridden.name].</span>"
+
+/datum/riding/proc/Process_Spacemove(direction)
+	if(ridden.has_gravity())
+		return 1
+
+	if(ridden.pulledby)
+		return 1
+
+	return 0
+
+/datum/riding/space/Process_Spacemove(direction)
+	return 1
+
+
+//atv
+/datum/riding/atv
+	keytype = /obj/item/key
+	generic_pixel_x = 0
+	generic_pixel_y = 4
+	vehicle_move_delay = 1
+	ridden = /obj/vehicle
```

Comment:
Setting these path defaults is confusing, they're not used anywhere as far as i can tell. Just leave them empty.